commit 35d36456fd327fc8c2a835dabc0b2f9fbf780680
Author: John Johansen <john.johansen@canonical.com>
Date:   Tue Sep 26 15:12:35 2017 -0400

    UBUNTU: SAUCE: LSM stacking: add support for stacking getpeersec_stream
    
    getpeersec_stream needs to use the "current" display LSM set by the
    prctl.
    
    Split out the getpeersec_stream implementation from the full stacking
    patch.
    
    Signed-off-by: John Johansen <john.johansen@canonical.com>
    Signed-off-by: Seth Forshee <seth.forshee@canonical.com>

diff --git a/security/security.c b/security/security.c
index b25b2e660e56..fee3a977cf0e 100644
--- a/security/security.c
+++ b/security/security.c
@@ -1982,8 +1982,21 @@ EXPORT_SYMBOL(security_sock_rcv_skb);
 int security_socket_getpeersec_stream(struct socket *sock, char __user *optval,
 				      int __user *optlen, unsigned len)
 {
+#ifdef CONFIG_SECURITY_STACKING
+	struct security_hook_list *hp;
+	char *lsm = lsm_of_task(current);
+
+	list_for_each_entry(hp, &security_hook_heads.socket_getpeersec_stream,
+									list) {
+		if (!lsm || !lsm[0] || !strcmp(lsm, hp->lsm))
+			return hp->hook.socket_getpeersec_stream(sock, optval,
+						optlen, len);
+	}
+	return -ENOPROTOOPT;
+#else
 	return call_int_hook(socket_getpeersec_stream, -ENOPROTOOPT, sock,
 				optval, optlen, len);
+#endif
 }
 
 int security_socket_getpeersec_dgram(struct socket *sock, struct sk_buff *skb, u32 *secid)
